cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -218,6 +229,12 @@ class ARROW_EXPORT S3FileSystem : public FileSystem {
   /// Return the actual region this filesystem connects to
   std::string region() const;
 
+  /// Set create_buckets property of options
```

Comment:
These trivial setters are not needed as people can access the attributes directly.

Reply:
`options()` returns a const reference; they could access but I don't think they could mutate right?

Being able to change this setting later felt important when I didn't support in `FromURI`, but maybe it's not as important now.

Reply:
> These trivial setters are not needed as people can access the attributes directly.

I'm not so sure that's true. This fails:
```cpp
  auto options = fs_->options();
  options.allow_bucket_creation = true;
  options.allow_bucket_deletion = true;

  ASSERT_EQ(fs_->options().allow_bucket_creation, true);
  ASSERT_EQ(fs_->options().allow_bucket_deletion, true);
```

Reply:
Okay, I think I'm fine at this point with only allowing setting them during construction, since we've added them to the URI parsing.